ccdproc/image_collection.py

```diff
@@ -686,6 +692,8 @@ def _generator(self, return_type,
             self._find_keywords_by_values(**kwd)
 
         ccd_kwargs = ccd_kwargs or {}
+        if 'hdu' not in ccd_kwargs.keys():
+            ccd_kwargs['hdu'] = extension_index
```

Comment:
you could also do `ccd_kwargs.setdefault('hdu', extension_index)` instead of checking if it's in the dictionary and adding it manually.

However this needs to be after line 705 (the one which defines the `extension_index`) otherwise you'll get a `NameError`.